File: app/src/main/java/org/cuieney/videolife/ui/act/MainActivity.java
package org.cuieney.videolife.ui.act;

import android.app.Activity;
import android.graphics.Color;
import android.util.Log;
import android.view.View;

import com.arlib.floatingsearchview.FloatingSearchView;
import com.ashokvarma.bottomnavigation.BottomNavigationBar;
import com.ashokvarma.bottomnavigation.BottomNavigationItem;
import com.jaeger.library.StatusBarUtil;

import org.cuieney.videolife.R;
import org.cuieney.videolife.common.base.BaseMainFragment;
import org.cuieney.videolife.common.base.SimpleActivity;
import org.cuieney.videolife.common.component.EventUtil;
import org.cuieney.videolife.common.utils.DelegatesExt;
import org.cuieney.videolife.kotlin.App;
import org.cuieney.videolife.kotlin.ui.fragment.video.VideoHomeFragment;
import org.cuieney.videolife.ui.fragment.essay.EssayFragment;
import org.cuieney.videolife.ui.fragment.music.MusicFragment;
import org.cuieney.videolife.ui.fragment.video.VideoFragment;
import org.greenrobot.eventbus.Subscribe;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import me.yokeyword.fragmentation.SupportFragment;

import static com.ashokvarma.bottomnavigation.BottomNavigationBar.BACKGROUND_STYLE_RIPPLE;
import static com.ashokvarma.bottomnavigation.BottomNavigationBar.MODE_FIXED;

public class MainActivity extends SimpleActivity implements BaseMainFragment.OnBackToFirstListener {


    private static final String TAG = "Main";
    @BindView(R.id.floating_search_view)
    FloatingSearchView mSearchView;
    @BindView(R.id.bottom_navigation_bar)
    BottomNavigationBar mNavigationView;

    List<SupportFragment> mFragments;

    @Override
    protected int getLayout() {
        return R.layout.design_layout;
    }

    @Override
    protected void initEventAndData() {
        mFragments = new ArrayList<>();
        mFragments.add(new VideoHomeFragment());
        mFragments.add(MusicFragment.newInstance());
        mFragments.add(new EssayFragment());
        mFragments.add(VideoFragment.newInstance());
        loadMultipleRootFragment(R.id.act_container, 0
                , mFragments.get(0)
                , mFragments.get(1)
                , mFragments.get(2)
                , mFragments.get(3)

        );

        initView();
    }


    private void initView() {
        mNavigationView
                .addItem(new BottomNavigationItem(R.drawable.movie_icon, "movie").setActiveColor("#6c4a41").setInActiveColor("#CCCCCC"))
                .addItem(new BottomNavigationItem(R.drawable.music_icon, "music").setActiveColor("#008867"))
                .addItem(new BottomNavigationItem(R.drawable.book_icon, "essay").setActiveColor("#8b6b64"))
                .addItem(new BottomNavigationItem(R.drawable.newspaper_icon, "newstand").setActiveColor("#485A66"))
                .initialise();
        mNavigationView.setBackgroundStyle(BACKGROUND_STYLE_RIPPLE);
        mNavigationView.setMode(MODE_FIXED);
        mNavigationView.setAutoHideEnabled(true);

        mNavigationView.setTabSelectedListener(new BottomNavigationBar.OnTabSelectedListener() {
            @Override
            public void onTabSelected(int position) {
                showHideFragment(mFragments.get(position));
                switch (position) {
                    case 0:
                        StatusBarUtil.setColor(MainActivity.this, Color.parseColor("#6c4a41"));
                        break;
                    case 1:
                        StatusBarUtil.setColor(MainActivity.this, Color.parseColor("#008867"));
                        break;
                    case 2:
                        StatusBarUtil.setColor(MainActivity.this, Color.parseColor("#8b6b64"));
                        break;
                    case 3:
                        StatusBarUtil.setColor(MainActivity.this, Color.parseColor("#485A66"));
                        break;
                }
            }

            @Override
            public void onTabUnselected(int position) {
            }

            @Override
            public void onTabReselected(int position) {
            }
        });

    }


    @Subscribe
    public void hide(String isHide) {
        if (isHide.equals("true")) {
            mSearchView.setVisibility(View.GONE);
        } else {
            mSearchView.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public void onStart() {
        super.onStart();
        EventUtil.register(this);
    }


    @Override
    public void onDestroy() {
        super.onDestroy();
        EventUtil.unregister(this);
    }

    @Override
    public void onBackToFirstFragment() {
        mNavigationView.selectTab(0);
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/ui/act/MainActivity.kt
package org.cuieney.videolife.kotlin.ui.act

import android.graphics.Color
import android.util.Log
import android.view.View
import com.ashokvarma.bottomnavigation.BottomNavigationBar
import com.ashokvarma.bottomnavigation.BottomNavigationBar.BACKGROUND_STYLE_RIPPLE
import com.ashokvarma.bottomnavigation.BottomNavigationBar.MODE_FIXED
import com.ashokvarma.bottomnavigation.BottomNavigationItem
import com.jaeger.library.StatusBarUtil
import org.cuieney.videolife.R
import kotlinx.android.synthetic.main.design_layout.*
import me.yokeyword.fragmentation.SupportFragment
import org.cuieney.videolife.common.component.EventUtil

import org.cuieney.videolife.common.component.RxBus
import org.cuieney.videolife.kotlin.base.BaseMainFragment
import org.cuieney.videolife.kotlin.base.SimpleActivity
import org.cuieney.videolife.kotlin.ui.fragment.essay.EssayFragment
import org.cuieney.videolife.kotlin.ui.fragment.music.MusicFragment
import org.cuieney.videolife.kotlin.ui.fragment.veer.VeerFragment
import org.cuieney.videolife.kotlin.ui.fragment.video.VideoFragment
import org.greenrobot.eventbus.Subscribe
import java.util.ArrayList

/**
 * Created by cuieney on 2017/6/2.
 */
open class MainActivity : SimpleActivity(), BaseMainFragment.OnBackToFirstListener {
    override val layout: Int = R.layout.design_layout


    lateinit var mFragments: ArrayList<SupportFragment>
    override fun initEventAndData() {
        mFragments = ArrayList<SupportFragment>()
        mFragments.add(VideoFragment())
        mFragments.add(MusicFragment())
        mFragments.add(EssayFragment())
        mFragments.add(VeerFragment())
        loadMultipleRootFragment(R.id.act_container, 0, mFragments[0], mFragments[1], mFragments[2], mFragments[3])

        initView()
    }


    private fun initView() {
        bottom_navigation_bar
                .addItem(BottomNavigationItem(R.drawable.movie_icon, "movie").setActiveColor("#6c4a41").setInActiveColor("#CCCCCC"))
                .addItem(BottomNavigationItem(R.drawable.music_icon, "music").setActiveColor("#008867"))
                .addItem(BottomNavigationItem(R.drawable.book_icon, "essay").setActiveColor("#8b6b64"))
                .addItem(BottomNavigationItem(R.drawable.panorama_icon, "panorama").setActiveColor("#485A66"))
                .initialise()
        bottom_navigation_bar.setBackgroundStyle(BACKGROUND_STYLE_RIPPLE)
        bottom_navigation_bar.setMode(MODE_FIXED)
        bottom_navigation_bar.setAutoHideEnabled(true)

        bottom_navigation_bar.setTabSelectedListener(object : BottomNavigationBar.OnTabSelectedListener {
            override fun onTabSelected(position: Int) {
                showHideFragment(mFragments[position])
                when (position) {
                    0 -> StatusBarUtil.setColor(this@MainActivity, Color.parseColor("#6c4a41"))
                    1 -> StatusBarUtil.setColor(this@MainActivity, Color.parseColor("#008867"))
                    2 -> StatusBarUtil.setColor(this@MainActivity, Color.parseColor("#8b6b64"))
                    3 -> StatusBarUtil.setColor(this@MainActivity, Color.parseColor("#485A66"))
                }
            }

            override fun onTabUnselected(position: Int) {}

            override fun onTabReselected(position: Int) {}
        })

    }


    @Subscribe
    fun hide(isHide: String) {
        if (isHide == "true") {
            floating_search_view.setVisibility(View.GONE)
        } else if(isHide == "false"){
            floating_search_view.setVisibility(View.VISIBLE)
        } else if(isHide == "tab_false"){
            floating_search_view.setVisibility(View.VISIBLE)
            bottom_navigation_bar.setVisibility(View.VISIBLE)
        }else if(isHide == "tab_true"){
            floating_search_view.setVisibility(View.GONE)
            bottom_navigation_bar.setVisibility(View.GONE)
        }
    }

    override fun onStart() {
        super.onStart()
        EventUtil.register(this)
    }


    override fun onDestroy() {
        super.onDestroy()
        EventUtil.unregister(this)
    }

    override fun onBackToFirstFragment() {
        bottom_navigation_bar.selectTab(0)
    }
}